akka-stream-tests/src/test/scala/akka/stream/scaladsl/WithContextUsageSpec.scala

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import scala.collection.immutable
+import akka.NotUsed
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.testkit.TestSubscriber.Probe
+import akka.stream.{ ActorMaterializer, ActorMaterializerSettings }
+import akka.stream.testkit.StreamSpec
+
+class WithContextUsageSpec extends StreamSpec {
+
+  val settings = ActorMaterializerSettings(system)
+  implicit val materializer = ActorMaterializer(settings)
+
+  "Context propagation used for committing offsets" must {
```

Comment:
Good set of additional tests. We should include some for the `mapConcat` + `grouped` scenario to make sure this is covered and also for `filter`.

Reply:
done